src/framework/components/camera/post-effect-pass.js

```diff
@@ -0,0 +1,436 @@
+                    if (uniforms2[k] === uname) {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        return false;
+    };
+
+    // collect global vars and return collisions with what's already in the list
+    var _collectGlobalTempVars = function(code, list) {
+        // Get code without any scoped stuff
+        var len = code.length;
+        var chr;
+        var scopeStart = 0;
+        var scopeEnd = 0;
+        var scopeDepth = 0;
+        var codeStart = 0;
+        var codeWithoutScopes = "";
+        var i, j;
+        for(i=0; i<len; i++) {
+            chr = code.charAt(i);
+            if (chr === "{") {
+                if (scopeDepth === 0) {
+                    scopeStart = i;
+                }
+                scopeDepth++;
+            } else if (chr === "}") {
+                if (scopeDepth === 1) {
+                    scopeEnd = i;
+                    codeWithoutScopes += code.substr(codeStart, (scopeStart - codeStart) + 1);
+                    codeStart = scopeEnd;
+                }
+                scopeDepth--;
+            }
+        }
+        codeWithoutScopes += code.substr(codeStart, (code.length - codeStart) + 1);
+        
+        // Find all global variable declarations and detect collisions
+        // ... won't work with re#defined types
+        var collisions = null;
+        var decls = codeWithoutScopes.match(/(float|int|bool|vec2|vec3|vec4|struct)([ \t\n\r]+[^\;]+[ \t\n\r]*\,*)+\;/g) || [];
+        var vars, varName;
+        for(i=0; i<decls.length; i++) {
+            vars = decls[i].split(",");
+            for(j=0; j<vars.length; j++) {
+                varName = vars[j].replace(/(float|int|bool|vec2|vec3|vec4|struct|\,|\;|\{|\})/g, "").trim();
+                if (list.indexOf(varName) >= 0) {
+                    if (!collisions) collisions = [];
+                    collisions.push(varName);
+                } else {
+                    list.push(varName);
+                }
+            }
+        }
+
+        // find all varying/uniform declarations (ideally should be possible to filter them out with first search...)
+         //and remove from list
+        var unrelevantDecls = codeWithoutScopes.match(/(uniform|varying|in|out)[ \t\n\r]+(float|int|bool|vec2|vec3|vec4|struct)([ \t\n\r]+[^\;]+[ \t\n\r]*\,*)+\;/g) || [];
+        var index;
+        for(i=0; i<unrelevantDecls.length; i++) {
+            vars = unrelevantDecls[i].split(",");
+            for(j=0; j<vars.length; j++) {
+                varName = vars[j].replace(/(float|int|bool|vec2|vec3|vec4|struct|uniform|varying|in|out|\,|\;|\{|\})/g, "").trim();
+                index = list.indexOf(varName);
+                if (index >= 0) {
+                    list.splice(index, 1);
+                }
+            }
+        }
+
+        return collisions;
+    };
+
+    /**
+     * @name pc.PostEffectPass
+     */
+    function PostEffectPass(script, options) {
```

Comment:
I think we should remove the script from here and maybe replace it with the `app`. And in options you can pass a `render` callback.

Reply:
Done.